### ISSUE TITLE
Increase minimum required PHP version

### ISSUE THREAD

**Issue by mindctrl** [easydigitaldownloads/easy-digital-downloads#6715]
We're going to start working towards increasing the minimum required version of PHP. Right now we're 5.2+.

`release/3.0` is currently set to 5.3+.

We need to move to at least 5.4+ for now.

EDD usage tracking does not combine the versions into an easy to read format. For example: `5.3.10-1ubuntu3.16`, `5.3.10-1ubuntu3.20`, `5.3.10-1ubuntu3.22`, and `5.3.10` are all tracked separately. I'll work on getting a good breakdown of each major version and update the issue with those stats.

EDD core PHP version stats as of 7/6/2018:
- 5.2 is at 0.98%
- 5.3 is at 4.79%
- 5.4 is at 13.64%
- 5.5 is at 9.79%
- 5.6 is at 44.78%
- 7.0 is at 18.84%
- 7.1 is at 6.09%
- 7.2 is at 1.10%

**Comment by mindctrl** [comment]
Updated with breakdown of major versions.

**Comment by cklosowski** [comment]
So WP Core just announced that their goal is to make Minimum version of PHP for WP in April of 2019 as PHP 5.6.

Depending on final release, we may want to look at this. I'm still prepping the email that will go out to sites reporting older versions of PHP. If we want to get ahead of the curve, we could possibly bump to 5.4 and give notices to all those users? Thoughts @easydigitaldownloads/core-team

**Comment by mintplugins** [comment]
I'm all for directing people to bump their PHP version more quickly. Most web hosts can do it pretty easily.

**Comment by SDavisMedia** [comment]
> I'm all for directing people to bump their PHP version more quickly. Most web hosts can do it pretty easily.

Same.

**Comment by spencerfinnell** [comment]
Unless the upcoming WordPress 5.x releases in the next couple of months really start freaking people out about dropping support then I doubt that 15k number is likely to change before April anyway.

**Comment by cklosowski** [comment]
Yeah, I'm not specifically talking 5.6, more opening up the conversation of with the announcement of Core going to 5.6, do we want to look at 5.4?

I'd even be fine with specifying this issue (renaming it) to be to bump to PHP 5.3, and then opening a new issue for 5.6 compatibility bump (to follow along with WP Core, when it happens).

**Comment by JJJ** [comment]
5.4 seems OK to me for now. Then we could publish a public timeline for our PHP bumps.

**Comment by DrewAPicture** [comment]
For what it's worth, the "Serve Happy" project being championed by the core PHP team will already be pushing messaging for upgrading PHP. I think we should follow core's lead and timeline, as painful as that's going to seem, for the simple fact that core is going to start throwing up roadblocks to anybody running < 5.6 in April and < 7.2 in November. We should get ahead of what core is doing, i.e. push core from behind instead of get pulled from ahead.

Simply put: We should bump the minimum to PHP 5.6 with 3.0, still allow stuff to work back to 5.3 without interruption, and when April rolls around get more insistent. Then around summer time, bump up to 7.2 and do the same push-pull routine again.

**Comment by chriscct7** [comment]
The November one only blocks PHP 5.6. It's just moving the minimum to PHP 7, not moving it to PHP 7.2. 

Also, it's currently suggested for no earlier than December. Unlike the April min raising, the December one is not guaranteed, and can/will slip.

**Comment by cklosowski** [comment]
Most recent stats:
5.2: 1% (878)
5.3: 4% (4,246)
5.4: 11% (12,479)
5.5: 7% (8,805)
5.6: 40% (47,673)
7.0: 19% (22,888)
7.1: 9% (10,050)
7.2: 8% (9,716)
7.3: 1% (1,090)

**Comment by cklosowski** [comment]
We can possibly go to 5.4+

```
SELECT count(distinct(url)) FROM edd_edd_tracking_sites
    WHERE last_checkin >= 2213472
    AND ( php_version like "5.2%" OR php_version LIKE "5.3%" );
```

Ends up with about 1.5% of reporting sites lower than PHP 5.4, just throwing in for discussion.

**Comment by cklosowski** [comment]
Quick update. Sites that have checked in to telemetry tracking within the last year using PHP 5.2 or 5.3, is down to 436. This is `0.81%` of sites running PHP versions lower than 5.4. I say it's time we go ahead and jump to a PHP 5.4 minimum version!

Thoughts here @easydigitaldownloads/developers

**Comment by ashleyfae** [comment]
I'm sad we're not going to at least 5.6, but otherwise seems fine to me. :P

**Comment by cklosowski** [comment]
@ashleyfae There is a chance we’ll go to 5.6. I need to have a few discussions about some upcoming plans, which may force us to go to 5.6 for the 3.0 release, meaning this notice will likely change, so I’m putting it on hold until I can talk to the people involved.

**Comment by robincornett** [comment]
Related:
- [WooCommerce 4.3 will set 7.2 as minimum PHP recommendation](https://wptavern.com/woocommerce-4-3-beta-available-for-testing)
- [WordPress bumps minimum PHP recommendation to 7.2](https://wptavern.com/wordpress-bumps-minimum-php-recommendation-to-7-2)

**Comment by ashleyfae** [comment]
Closing, as the minimum has been increased to 5.6 already.
